workflow/controller/pod_cleanup.go

```diff
@@ -18,7 +18,7 @@ func (woc *wfOperationCtx) queuePodsForCleanup() {
 	for _, obj := range objs {
 		pod := obj.(*apiv1.Pod)
 		nodeID := woc.nodeID(pod)
-		if !woc.execWf.Status.Nodes[nodeID].Fulfilled() {
+		if !woc.execWf.Status.Nodes[nodeID].Phase.Fulfilled() {
```

Comment:
not just Completed()?

Reply:
Fulfilled will take care skipped and Omitted nodes also.  Fulfilled is right one
```go
// Fulfilled returns whether a phase is fulfilled, i.e. it completed execution or was skipped or omitted
func (phase NodePhase) Fulfilled() bool {
	return phase.Completed() || phase == NodeSkipped || phase == NodeOmitted
}
```

Reply:
strange they are inconsistent